Build failure: zig-rlp dependency uses removed std.mem.copy (Zig 0.14)

## Description

When attempting to build zigeth with Zig 0.14.1, the build fails because the `zig-rlp` dependency uses `std.mem.copy`, which was removed in Zig 0.14.

## Environment

- Zig version: 0.14.1
- OS: macOS (darwin) / Linux (Ubuntu 22.04 in Docker)

## Steps to Reproduce

```bash
mkdir test-zigeth && cd test-zigeth
zig init
zig fetch --save=zigeth git+https://github.com/ch4r10t33r/zigeth#main
# Add zigeth as dependency in build.zig
zig build
```

## Error

```
.zig-cache/p/zig-rlp-0.0.1-CKmuSxSz7ZSfnPl-/src/rlp.zig:16:21: error: no field or decl named 'copy' in 'mem'
                std.mem.copy(u8, self.data[self.pos..], slice);
                ~~~~~~~~^~~~
```

## Root Cause

The `zig-rlp` dependency (pinned at `CKmuSxSz7ZSfnPl-...`) uses `std.mem.copy`, which was deprecated and then removed in Zig 0.14. The replacement is `@memcpy`.

## Suggested Fix

Either:
1. Update the `zig-rlp` dependency to a version that uses `@memcpy` instead of `std.mem.copy`
2. Or update `build.zig.zon` to point to a newer commit of `zig-rlp` that has the fix

## Additional Context

I was attempting to integrate zigeth into a native rollup sequencer project for L1 blockchain interactions (RPC client, transaction signing, etc.). Due to this build failure, I had to fall back to using Zig's standard library `std.http.Client` directly for HTTP/HTTPS RPC calls.

The zigeth library would be very useful for Ethereum development in Zig - looking forward to seeing this fixed!

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -20,7 +20,7 @@ jobs:
       - name: Setup Zig
         uses: goto-bus-stop/setup-zig@v2
         with:
-          version: 0.14.1
+          version: 0.15.0
 
       - name: Create cache directory
         run: mkdir -p .zig-cache/global
@@ -49,7 +49,7 @@ jobs:
       - name: Setup Zig
         uses: goto-bus-stop/setup-zig@v2
         with:
-          version: 0.14.1
+          version: 0.15.0
 
       - name: Create cache directories (Unix)
         if: runner.os != 'Windows'
@@ -96,7 +96,7 @@ jobs:
       - name: Setup Zig
         uses: goto-bus-stop/setup-zig@v2
         with:
-          version: 0.14.1
+          version: 0.15.0
 
       - name: Create cache directories (Unix)
         if: runner.os != 'Windows'
@@ -134,7 +134,7 @@ jobs:
       - name: Setup Zig
         uses: goto-bus-stop/setup-zig@v2
         with:
-          version: 0.14.1
+          version: 0.15.0
 
       - name: Create cache directory
         run: mkdir -p .zig-cache/global
@@ -160,7 +160,7 @@ jobs:
       - name: Setup Zig
         uses: goto-bus-stop/setup-zig@v2
         with:
-          version: 0.14.1
+          version: 0.15.0
 
       - name: Create cache directory
         run: mkdir -p .zig-cache/global

--- a/build.zig.zon
+++ b/build.zig.zon
@@ -16,7 +16,7 @@
     .dependencies = .{
         .zig_eth_secp256k1 = .{
             .url = "https://github.com/DaviRain-Su/zig-eth-secp256k1/archive/refs/heads/main.tar.gz",
-            .hash = "zig_eth_secp256k1-0.1.0-_U97sLbpDAATA_3x8cFERYszQ8DPzDck3ZiffzwQOcnt",
+            .hash = "zig_eth_secp256k1-0.1.0-_U97sBjqDACdmkjL_I_1dncMBvOZ0iSxbcMg4pIcVM4P",
         },
     },
     .paths = .{

